stages/org.osbuild.ostree.deploy.container

```diff
@@ -0,0 +1,147 @@
+"""
+
+import os
+import sys
+
+import osbuild.api
+from osbuild.util import containers, ostree
+from osbuild.util.mnt import MountGuard
+
+CAPABILITIES = ["CAP_MAC_ADMIN"]
+
+SCHEMA_2 = """
+"options": {
+  "additionalProperties": false,
+  "required": ["osname"],
+  "properties": {
+    "mounts": {
+      "description": "Mount points of the final file system",
+      "type": "array",
+      "items": {
+        "description": "Description of one mount point",
+        "type": "string"
+      }
+    },
+    "osname": {
+      "description": "Name of the stateroot to be used in the deployment",
+      "type": "string"
+    },
+    "kernel_opts": {
+      "description": "Additional kernel command line options",
+      "type": "array",
+      "items": {
+        "description": "A single kernel command line option",
+        "type": "string"
+      }
+    },
+    "target_imgref": {
+      "description": "imageref used as the source of truth for updates",
+      "type": "string"
+    },
+    "rootfs": {
+      "description": "Identifier to locate the root file system",
+      "type": "object",
+      "oneOf": [{
+        "required": ["uuid"]
+      }, {
+        "required": ["label"]
+      }],
+      "properties": {
+        "label": {
+          "description": "Identify the root file system by label",
+          "type": "string"
+        },
+        "uuid": {
+          "description": "Identify the root file system by UUID",
+          "type": "string"
+        }
+      }
+    }
+  }
+},
+"inputs": {
+  "type": "object",
+  "additionalProperties": false,
+  "required": ["images"],
+  "properties": {
+    "images": {
+      "type": "object",
+      "description": "Container Image to deploy",
+      "additionalProperties": true
+    },
+    "manifest-lists": {
+      "type": "object",
+      "description": "Optional manifest lists to merge into images. The metadata must specify an image ID to merge to.",
+      "additionalProperties": true
+    }
+  }
+}
+"""
+
+
+def make_fs_identifier(desc):
+    for key in ["uuid", "label"]:
+        val = desc.get(key)
+        if val:
+            return f"{key.upper()}={val}"
+    raise ValueError("unknown rootfs type")
+
+
+def ostree_container_deploy(tree, inputs, osname, target_imgref, kopts):
+    images = containers.parse_containers_input(inputs)
+    for image in images.values():
+        with containers.container_source(image) as (image_name, image_source):
+            extra_args = []
+            imgref = f"ostree-unverified-image:{image_source}"
+
+            extra_args.append(f'--imgref={imgref}')
+            extra_args.append(f'--stateroot={osname}')
+
+            # consider implicit signature verification type checks, but 
+            # can't think of a "clean" way to do it yet other than 
+            # parsing the target-imgref and separating by the ':' character
+            if target_imgref:
+                extra_args.append(f'--target-imgref={target_imgref}')
```

Comment:
So by default we wouldn't set the target_imgref to anything unless the user told us to?

Reply:
I think we should either:

default to setting it to `ostree-unverified-registry:{image_name}` or make it required in the schema.

I think Colin was advocating for not hardcoding `ostree-unverified-registry` anywhere, so that would lead us to the 2nd option.

Reply:
I also had the thought of making it more similar to how we did it before (ie. in `options`, explicitly stating remote, the verification type, taking image_name from the container image input), but I was considering the last paragraph of Achilleas' comment: https://github.com/osbuild/osbuild/pull/1402#discussion_r1364154959. Do you have any thoughts on this?

Reply:
> make it required in the schema.
Oh wait sorry, I didn't realize you meant make `target_imgref` required. Yeah, I cool with adding that in.